### PR TITLE
YANG model for enabling fabric monitor attribute

### PR DESCRIPTION
**Pull request by jfeng-arista**
#### Why I did it
The new added attributes related fabric link monitoring are missing YANG model. Adding them in this change. 

For fabric monitor table: 
> monCapacityThreshWarn
> monState

For fabric port:
> forceUnisolateStatus 
##### Work item tracking
- Microsoft ADO **(number only)**:

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang [hunk 1]
```diff
@@ -43,6 +43,19 @@ module sonic-fabric-monitor{
...
 
+                leaf monCapacityThreshWarn {
+                    type uint32;
+                    default 10;
+                    description "Percentage of up fabric links.";
+                }
+
+                leaf monState {
+                    description "Configuration to set fabric link monitoring state: enable/disable";
+                    type string {
```
Comment (arlakshm) [inline src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang:54]: if the acceptable values are `enable/disable` can we make the type as enum or pattern?

File: src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang [hunk 2]
```diff
@@ -43,6 +43,19 @@ module sonic-fabric-monitor{
                     description "Consecutive polls with lesser error rate for inclusion.";
                 }
 
+                leaf monCapacityThreshWarn {
+                    type uint32;
```
Comment (arlakshm) [inline src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang:47]: the max capacity is 100%, do we need uint32 for this?

File: src/sonic-yang-models/yang-models/sonic-fabric-port.yang [hunk 3]
```diff
@@ -52,6 +52,12 @@ module sonic-fabric-port{
                     }
                 }
 
+                leaf forceUnisolateStatus {
```
Comment (arlakshm) [inline src/sonic-yang-models/yang-models/sonic-fabric-port.yang:55]: can we make this as Enum to restrict the acceptable values 0 or 1?

**Review by arlakshm — COMMENTED** [review]
please add some UT

File: src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang (hunk 2 above)
Comment (jfeng-arista) [reply src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang:47]: > the max capacity is 100%, do we need uint32 for this?

revised.

**Comment by jfeng-arista** [comment]
ASSERT TRUE FAILED(5085): sizeof(sai_switch_health_data_t) == (SAI_SWITCH_HEALTH_DATA_T_SIZE): wrong size of sai_switch_health_data_t, expected 16, got 4
make[2]: *** [Makefile:89: all] Error 1
make[2]: Leaving directory '/sonic/src/dash-sai/DASH/dash-pipeline/SAI/SAI/meta'

does not seem related to our code

**Comment by jfeng-arista** [comment]
I see the same error on other PRs as well, thinking this is not related the change, try one more time of runming the pipleline, otherwise, we will wait for the issue get fixed upstream

**Comment by jfeng-arista** [comment]
same failure before, I will wait a bit for the upstream fix then kick off the test pipeline again

**Comment by kenneth-arista** [comment]
@arlakshm @wenyiz2021 can you help unblock the `ms_checker` failure? We don't have permission to see what is failing.

**Review by vmittal-msft — APPROVED** [review]

File: src/sonic-yang-models/yang-models/sonic-fabric-port.yang (hunk 3 above)
Comment (arlakshm) [reply src/sonic-yang-models/yang-models/sonic-fabric-port.yang:55]: can you close this comment?

**Comment by abdosi** [comment]
@qiluo-msft : Can you please help to merge this.

**Comment by jfeng-arista** [comment]
> please add some UT

I updated existing UT test test_fabric_port_table and test_fabric_monitor_data_table to cover the new filed. please let me know if this is enough. thank you

**Comment by rlhui** [comment]
@arlakshm  and @mlok-nokia  please quickly review so this one could be merged, thanks.

**Review by arlakshm — DISMISSED** [review]

**Comment by rlhui** [comment]
@qiluo-msft , please help review this quickly? thanks.

File: src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang [hunk 4]
```diff
@@ -32,17 +32,31 @@ module sonic-fabric-monitor{
                 }
 
                 leaf monPollThreshIsolation {
-                    type uint32;
+                    type uint8;
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang:35]: why do you change type here?

File: src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang (hunk 1 above)
Comment (wen587) [reply src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang:54]: If only accept enable/disable, we can remove length?

File: src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang (hunk 4 above)
Comment (wen587) [reply src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang:35]: same question also on `leaf monPollThreshRecovery {`

File: src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang (hunk 1 above)
Comment (wen587) [reply src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang:54]: And can be used by type stypes:mode-status;
Check example here: https://github.com/sonic-net/sonic-buildimage/blob/cab94964d94381ac399361436e85d6ca6637cbd9/src/sonic-yang-models/yang-models/sonic-interface.yang#L80

File: src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang [hunk 5]
```diff
@@ -32,17 +32,31 @@ module sonic-fabric-monitor{
...
 
+                leaf monCapacityThreshWarn {
+                    type uint8;
+                    default 10;
+                    description "Percentage of up fabric links.";
+                }
+
+                leaf monState {
+                    description "Configuration to set fabric link monitoring state: enable/disable";
+                    type string {
```
Comment (wen587) [inline src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang:54]: If only accept enable/disable, we can remove length?
And can be used by type stypes:mode-status;
Check example here: https://github.com/sonic-net/sonic-buildimage/blob/cab94964d94381ac399361436e85d6ca6637cbd9/src/sonic-yang-models/yang-models/sonic-interface.yang#L80

File: src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang (hunk 4 above)
Comment (jfeng-arista) [reply src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang:35]: > why do you change type here?
I am expecting this number be very small, like 1,2, or something, so I changed it to uint8 . Let me know if you prefer leave it as orginal uint32.

File: src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang (hunk 5 above)
Comment (jfeng-arista) [reply src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang:54]: > sonic-buildimage/src/sonic-yang-models/yang-models/sonic-interface.yang

got it, revised based on your suggestion.

File: src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang (hunk 4 above)
Comment (ganglyu) [reply src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang:35]: If the number is really small, can you add range?

Comment (jfeng-arista) [reply src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang]: > If the number is really small, can you add range?

added

File: src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang [hunk 6]
```diff
@@ -32,17 +32,35 @@ module sonic-fabric-monitor{
...
                     default 8;
                     description "Consecutive polls with lesser error rate for inclusion.";
                 }
 
+                leaf monCapacityThreshWarn {
+                    type uint8 {
+                        range 5..250;
+                    }
+                    default 10;
+                    description "Percentage of up fabric links.";
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang:55]: Why is this percentage greater than 100?

Comment (jfeng-arista) [reply src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang:55]: > Why is this percentage greater than 100?

It was from comparing with other needs, but I forgot we changed design ,  fixed it

Comment (ganglyu) [reply src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang:55]: Will you change the range or description?

**Review by ganglyu — APPROVED** [review]
LGTM

**Comment by jfeng-arista** [comment]
> LGTM

thanks

File: src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang [hunk 7]
```diff
@@ -32,17 +32,35 @@ module sonic-fabric-monitor{
...
+                    type uint8 {
+                        range 5..100;
+                    }
+                    default 10;
+                    description "Percentage of up fabric links.";
+                }
+
+                leaf monState {
+                    description "Configuration to set fabric link monitoring state: enable/disable";
+                    type stypes:mode-status;
```
Comment (wen587) [inline src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang:60]: import is also needed:
https://github.com/sonic-net/sonic-buildimage/blob/cab94964d94381ac399361436e85d6ca6637cbd9/src/sonic-yang-models/yang-models/sonic-interface.yang#L8-L10

Comment (jfeng-arista) [reply src/sonic-yang-models/yang-models/sonic-fabric-monitor.yang:64]: > import is also needed:
> 
> https://github.com/sonic-net/sonic-buildimage/blob/cab94964d94381ac399361436e85d6ca6637cbd9/src/sonic-yang-models/yang-models/sonic-interface.yang#L8-L10

done

**Comment by kenneth-arista** [comment]
Referencing https://github.com/sonic-net/sonic-utilities/issues/3443

**Review by wen587 — APPROVED** [review]

**Comment by jfeng-arista** [comment]
"Error type: DUTHOST_UNREACHABLE
Error message: DUT host unreachable for vms-kvm-dual-t0 on running test_pretest.py|||vms-kvm-dual-t0_215703
"

it failed with above issue. 

kick off another build

**Comment by jfeng-arista** [comment]
Error type: START_WORKER_AGENT_FAILED
Error message: Prepare testbed failed: testbed_q_sonic-elastictest-prod-vmss-D8s-v3_216742 for START_WORKER_AGENT_FAILED, testbed_id is: 66da046a066cc8cdf5ad59eb, cmd is: Start sonic-mgmt worker agent
Operation failed with exception: Exception('Test plan id: 66da01891b0209cf5f97a1fb, status: None, result: FAILED, Elapsed 0 seconds. Check https://elastictest.org/scheduler/testplan/66da01891b0209cf5f97a1fb for test plan status')


The failure does not look related to the change

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**